docker.go

```diff
@@ -265,6 +278,79 @@ func (c *DockerContainer) Exec(ctx context.Context, cmd []string) (int, error) {
 	return exitCode, nil
 }
 
+func (d *DockerContainer) StartLogProducer(ctx context.Context) error {
+	go func() {
+		options := types.ContainerLogsOptions{
+			ShowStdout: true,
+			ShowStderr: true,
+			Follow:     true,
+		}
+
+		ctx, cancel := context.WithTimeout(ctx, time.Second*5)
+		defer cancel()
+
+		r, err := d.provider.client.ContainerLogs(ctx, d.GetContainerID(), options)
+		if err != nil {
+			// if we can't get the logs, panic, we can't return an error to anything
+			// from within this goroutine
+			panic(err)
+		}
+
+		for {
+			select {
+			case <-d.stopProducer:
+				err := r.Close()
+				if err != nil {
+					// we can't close the read closer, this should never happen
+					panic(err)
+				}
+				return
+			default:
+				h := make([]byte, 8)
+				_, err := r.Read(h)
+				if err != nil {
+					// this explicitly ignores errors
+					// because we want to keep procesing even if one of our reads fails
+					continue
+				}
+
+				count := binary.BigEndian.Uint32(h[4:])
+				if count == 0 {
+					continue
+				}
+				logType := h[0]
+				if logType > 2 {
+					panic(fmt.Sprintf("received inavlid log type: %d", logType))
+				}
+
+				// a map of the log type --> int representation in the header, notice the first is blank, this is stdin, but the go docker client doesn't allow following that in logs
+				logTypes := []string{"", StdoutLog, StderrLog}
+
+				b := make([]byte, count)
+				_, err = r.Read(b)
+				if err != nil {
+					fmt.Fprintf(os.Stderr, "error occurred reading log with known length %s", err.Error())
```

Comment:
This looks like something that we can solve as soon as we will introduce a logger.
Should we place a `TODO` here about it?

Reply:
yes good idea @gianarb , I can add that here, is there any sort of format we should follow?  so we can search for all places where we should add a logger?

ex. `TODO: add-logger: <description>`

Reply:
that way, when we do add a logger, we can search for "TODO: add-logger:" and find all places to add it

Reply:
Yes!